integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -1,77 +1,261 @@
+ */
 public class DistroXEncryptedVolumeTest extends AbstractE2ETest {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(DistroXEncryptedVolumeTest.class);
+
+    public String environmentName;
+
     @Inject
     private DistroXTestClient distroXTestClient;
 
     @Inject
-    private TestParameter testParameter;
+    private SshEnaDriverCheckActions sshEnaDriverCheckActions;
 
     @Inject
-    private SshEnaDriverCheckActions sshEnaDriverCheckActions;
+    private CredentialTestClient credentialTestClient;
+
+    @Inject
+    private EnvironmentTestClient environmentTestClient;
+
+    @Inject
+    private AwsCloudProvider awsCloudProvider;
+
+    @Inject
+    private AzureCloudProvider azureCloudProvider;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private FreeIpaInstanceUtil freeIpaInstanceUtil;
+
+    @Inject
+    private SdxUtil sdxUtil;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
 
     @Override
     protected void setupTest(TestContext testContext) {
+        notSupportedCloudPlatform(CloudPlatform.GCP);
         testContext.getCloudProvider().getCloudFunctionality().cloudStorageInitialize();
         createDefaultUser(testContext);
         initializeDefaultBlueprints(testContext);
         createDefaultCredential(testContext);
-        createEnvironmentWithFreeIpaAndDatalake(testContext);
     }
 
-    @Test(dataProvider = TEST_CONTEXT)
+    @Test(dataProvider = TEST_CONTEXT, description = "KNOWN ISSUE: CB-15601 Azure Disk Encryption Set creation throws" +
+            " PSQLException if environment name too long")
     @UseSpotInstances
     @Description(
             given = "there is a running cloudbreak",
-            when = "a valid DistroX create request is sent with encrypted discs",
-            then = "DistroX cluster is created")
+            when = "create an environment then a valid distroX with disk encryption",
+            then = "freeIpa, sdx and distroX volumes should be encrypted with provided keys")
     public void testCreateDistroXWithEncryptedVolume(TestContext testContext) {
-        String externalDatabaseName = resourcePropertyProvider().getName();
+        SdxDatabaseRequest sdxDatabaseRequest = new SdxDatabaseRequest();
+        DistroXDatabaseRequest distroXDatabaseRequest = new DistroXDatabaseRequest();
         List<DistroXInstanceGroupTestDto> distroXInstanceGroupTestDtos = new DistroXInstanceGroupsBuilder(testContext)
                 .defaultHostGroup()
                 .withDiskEncryption()
                 .build();
-        testContext.given(externalDatabaseName, DistroXExternalDatabaseTestDto.class)
-                .withAvailabilityType(DistroXDatabaseAvailabilityType.NON_HA)
+        sdxDatabaseRequest.setAvailabilityType(SdxDatabaseAvailabilityType.NON_HA);
+        distroXDatabaseRequest.setAvailabilityType(DistroXDatabaseAvailabilityType.NON_HA);
```

Comment:
I wonder if an external DB is indeed warranted in this test. It would be, of course, if RDS encryption settings were also verified, but I cannot see this happening.